spec/controllers/api/openid_connect/authorizations_controller_spec.rb

```diff
@@ -380,12 +380,71 @@
         end
       end
     end
+
+    context "ensure public:read always there" do
+      it "added with only openid" do
+        get new_api_openid_connect_authorization_path, params: {client_id:     client.client_id,
+                                                                redirect_uri:  "http://localhost:3000/",
+                                                                response_type: "id_token token",
+                                                                scope:         "openid",
+                                                                nonce:         418_093_098_3,
+                                                                state:         418_093_098_3}
+        post api_openid_connect_authorizations_path, params: {approve: "true"}
+        expect(alice.reload.authorizations.first.scopes).to include("public:read")
+      end
+    end
+
+    context "with contacts:read and private linkage requirement" do
+      it "fails without contacts:read on private:read" do
+        scopes = "openid private:read"
+        get new_api_openid_connect_authorization_path, params: {client_id:     client.client_id,
+                                                                redirect_uri:  "http://localhost:3000/",
+                                                                response_type: "id_token token",
+                                                                scope:         scopes,
+                                                                nonce:         418_093_098_3,
+                                                                state:         418_093_098_3}
+        expect(response.status).to eq(302)
```

Comment:
I believe the redirect on errors is coming from the same place as the redirect on success.  The client app request sends the information to the middle-server for further processing in both cases.  So this would give that server the chance to log the error, etc.